src/test/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketPluginConfigurationTest.java

```diff
@@ -72,15 +80,10 @@ public void setup() {
 
     @Test
     public void testConfigureChangedBaseUrlUpdatesJob() throws Exception {
-        FreeStyleProject freeStyleProject = jenkins.createFreeStyleProject();
-        BitbucketSCM bitbucketSCMInitial = mock(BitbucketSCM.class);
-        when(bitbucketSCMInitial.getServerId()).thenReturn("0");
-        freeStyleProject.setScm(bitbucketSCMInitial);
-        assertThat(jenkins.getInstance().getAllItems(FreeStyleProject.class).get(0).getScm(),
-                equalTo(bitbucketSCMInitial));
+        BitbucketSCM initialSCM = mock(BitbucketSCM.class);
```

Comment:
Also related to the discussion of unit vs. integration tests: if this stays an integration tests (as opposed to converting it to a proper unit test), which to me seems like the right choice, we might wanna create a real Bitbucket SCM using the `BitbucketJenkinsRule`, like in `BitbucketProjectConfigurationIT`.

Reply:
P.S. I know it's not strictly related to what this PR is trying to fix, but just as an opportunistic refactor (completely optional in this PR).